src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4170,7 +4218,7 @@ private TypeSymbol VisitArrayInitialization(TypeSymbol type, BoundArrayInitializ
                     expressionsNoConversions.Add(expressionNoConversion);
                     conversions.Add(conversion);
                     SnapshotWalkerThroughConversionGroup(expression, expressionNoConversion);
-                    var expressionType = VisitRvalueWithState(expressionNoConversion);
+                    var expressionType = VisitRvalueInConversion(expressionNoConversion);
```

Comment:
Does this need to be conditional on the presence of a conversion?
This question applies to every place that makes use of the new APIs. Alternatively, can we assert that there must have been a conversion?

Reply:
I think it's actually asserted by the DebugVerifier. If we call `VisitRvalueInConversion`, the lambda won't be visited by the `public VisitLambda`, and then if we fail to call `VisitConversion`, the lambda won't be visited by the `private VisitLambda` either, hence DebugVerifier will report unvisited nodes inside the lambda's body.